types/tabl-service/index.d.ts

```diff
@@ -0,0 +1,216 @@
+     * @returns A Promise that resolves once the Stripe object has been successfully instantiated.
+     */
+    initStripe: (stripeAccount: string) => Promise<void>;
+
+    /**
+     * Sets the loading state of the application or a specific component, optionally displaying a loading message.
+     * This can be used to provide visual feedback to the user during operations that require waiting, such as data fetching or processing.
+     *
+     * @param shouldLoad - A boolean indicating whether the loading state should be activated (`true`) or deactivated (`false`).
+     * @param message - An optional string that provides a contextual message to display during the loading state.
+     */
+    setLoading: (shouldLoad: boolean, message?: string) => void;
+
+    /**
+     * Displays a toast notification with a specified message and interface using the `ToastType` enumeration.
+     * This function is utilized to provide user feedback in various scenarios, such as operation success, warnings, errors, or informational messages.
+     * The `type` parameter dictates the visual styling and possibly the iconography of the toast, aligning it with the nature of the message.
+     *
+     * @param params - An object containing the optional `message` to be displayed and the `type` of the toast.
+     * @param params.message - An optional string representing the message to be shown in the toast. If not provided, the toast may display a default message based on the `type`.
+     * @param params.type - A `ToastType` value that specifies the toast's category, influencing its visual representation and the context in which it should be used.
+     */
+    showToast: ({ message, type }: { message?: string; type: ToastType }) => void;
+
+    /**
+     * Displays a custom alert dialog with a specified title, text content, and a button.
+     * This function is designed to provide a standardized method for showing alert messages to the user, with the flexibility to indicate the nature of the alert through a interface parameter.
+     * The `type` parameter allows for basic customization of the alert's appearance or behavior, typically by changing colors or icons to reflect an error or success state.
+     *
+     * @param title - The title of the alert dialog, intended to summarize the nature of the alert or to grab the user's attention.
+     * @param text - The detailed text content of the alert, providing more information about the alert or instructions for the user.
+     * @param buttonText - The text label for the alert dialog's button, which the user can click to acknowledge and dismiss the alert.
+     * @param type - A string that specifies the alert's type, with accepted values being 'error' or 'success'. This affects the visual styling of the alert to match the context of the message.
+     */
+    showCustomAlert: (title: string, text: string, buttonText: string, type: "error" | "success") => void;
+}
+
+export enum ToastType {
```

Comment:
This being exported means this is a module, but the npm package `tabl-service` doesn't exist, so how are you expecting this to be imported? Is this supposed to be defining a global?

It's exceedingly rare to have a nonNpm package export something.

Reply:
If this is for an internal package, it definitely should not be on DT. DT is for public npm packages, global types, runtime environment descriptions, etc.

You can publish a package like this internally, and if needed add the package to the `types` array in `tsconfig` to include it (or use references), but it definitely should not be on DT if the public cannot use it.